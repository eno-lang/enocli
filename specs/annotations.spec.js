const path = require('path');

const analyze = require('../lib/analysis/analyze.js');

const DOCUMENT_PATH = path.join(__dirname, 'fixtures/annotations.eno');

describe('analyze', () => {
  it('parses and acts on annotation comments', () => {
    const astDocument = analyze(DOCUMENT_PATH);

    expect(astDocument).toMatchSnapshot();
  });
});
