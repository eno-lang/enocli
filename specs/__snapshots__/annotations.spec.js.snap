// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analyze parses and acts on annotation comments 1`] = `
Object {
  "elementRequired": true,
  "elementType": "section",
  "elements": Array [
    Object {
      "elementRequired": false,
      "elementType": "empty",
      "internalKey": "my_empty_key",
      "key": "empty",
      "parent": [Circular],
    },
    Object {
      "elementRequired": false,
      "elementType": "field",
      "internalKey": "my_field_key",
      "key": "field",
      "parent": [Circular],
      "valueRequired": false,
      "valueTypes": Array [
        "integer",
      ],
    },
    Object {
      "elementRequired": false,
      "elementType": "fieldset",
      "entries": Array [
        Object {
          "elementRequired": false,
          "elementType": "entry",
          "internalKey": "my_entry_key",
          "key": "entry",
          "parent": [Circular],
          "valueRequired": false,
          "valueTypes": Array [
            "integer",
          ],
        },
      ],
      "internalKey": "my_fieldset_key",
      "key": "fieldset",
      "parent": [Circular],
    },
    Object {
      "elementRequired": false,
      "elementType": "list",
      "internalKey": "my_list_key",
      "key": "list",
      "parent": [Circular],
      "valueTypes": Array [
        "integer",
      ],
      "valuesRequired": false,
    },
    Object {
      "elementRequired": false,
      "elementType": "section",
      "elements": Array [
        Object {
          "elementRequired": false,
          "elementType": "field",
          "internalKey": "my_field_key",
          "key": "field",
          "parent": [Circular],
          "valueRequired": false,
          "valueTypes": Array [
            "integer",
          ],
        },
      ],
      "internalKey": "my_section_key",
      "key": "section",
      "parent": [Circular],
    },
  ],
  "internalKey": "document",
}
`;
